### PR TITLE
Met 75 check all todos fixme in the project

### PR DESCRIPTION
**Pull request by afmika**
This partially solves MET-75 (TODO / FIXME)

### CODE REVIEW

File: typegate/deno_import.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "oauth2_client": "https://deno.land/x/oauth2_client@v0.2.1/mod.ts",
     "test/mock_fetch": "https://deno.land/x/mock_fetch@0.3.0/mod.ts",
     "json_schema_typed": "https://deno.land/x/json_schema_typed@v8.0.0/draft_latest.ts",
-    "url_join": "https://deno.land/x/url_join@1.0.0/mod.ts"
+    "url_join": "https://deno.land/x/url_join@1.0.0/mod.ts",
+    "collections": "https://deno.land/std@0.139.0/collections/mod.ts"
```
Comment (Natoandro) [inline typegate/deno_import.json:24]: We already have `"std/"` on line 3; or do we need features from that specific (old) version?

Comment (Natoandro) [reply typegate/deno_import.json:24]: Imports could be:
```ts
import { mapValues } from "std/collections/map_values.ts";
```
This uses the version specified on the `"std/"` field, and download less files: only `map_values.ts` and its deps; `"collections/mod.ts"` would download all the imports in `mod.ts`.

File: typegate/src/engine.ts [hunk 2]
```diff
@@ -31,6 +31,8 @@ import { Planner } from "./planner/mod.ts";
 import { FromVars } from "./runtimes/graphql.ts";
 import config from "./config.ts";
 import * as semver from "std/semver/mod.ts";
+import { mapValues } from "collections";
```
Comment (Natoandro) [inline typegate/src/engine.ts:34]: cf supra

File: typegate/src/runtimes/random.ts [hunk 3]
```diff
@@ -60,33 +71,45 @@ export class RandomRuntime extends Runtime {
...
+              return this.chance.email();
+            }
+            return this.chance.string();
+          case "boolean":
+            return this.chance.bool();
+          case "array": {
+            const res = [];
+            let size = this.chance.integer({ min: 1, max: 10 });
+            while (size--) {
+              const childNodeName = this.getTgTypeNameByIndex(typ.items);
```
Comment (Natoandro) [inline typegate/src/runtimes/random.ts:102]: Calls `this.getTgTypeNameByIndex` on each iteration.

File: libs/xtask/src/codegen/typescript.rs [hunk 4]
```diff
@@ -100,6 +100,11 @@ pub fn run() -> Result<()> {
 }
 
 fn print_module<W: Write>(cm: Lrc<SourceMap>, module: &Module, writer: W) -> Result<()> {
+    // ref: https://doc.rust-lang.org/std/env/consts/constant.OS.html
```
Comment (Natoandro) [inline libs/xtask/src/codegen/typescript.rs:103]: Though not critical, a compile type check would be more efficient.
```rust
#[cfg(target_os = "windows")]
let new_line = "\r\n";
#[cfg(not(target_os = "windows"))]
let new_line = "\n";
```

**Review by Natoandro — CHANGES_REQUESTED** [review]

File: typegate/src/runtimes/random.ts (hunk 3 above)
Comment (afmika) [reply typegate/src/runtimes/random.ts:102]: You are right, I will refactor fact.

**Review by Natoandro — DISMISSED** [review]

**Review by Natoandro — DISMISSED** [review]

**Review by Natoandro — APPROVED** [review]

**Pull request merged**